base/channels.jl

```diff
@@ -129,12 +135,13 @@ julia> String(collect(chnl))
 "hello world"
 ```
 """
-function Channel{T}(func::Function, size=0; taskref=nothing, spawn=false) where T
+function Channel{T}(func::Function, size=0; taskref=nothing, spawn=false, threadpool=:default) where T
```

Comment:
What should we do if someone sets `threadpool=:interactive` or `threadpool=:default` but forgets to set `spawn=true`? Right now, it'll be silently ignored, which isn't very user-friendly.

An alternative could be to default to `threadpool=nothing`, which allows us to detect _whether it was explicitly set by the caller_, and if it's !=nothing, we either:
- automatically set spawn=true (i think i'm in favor of this option), since it's obvious that's what they meant,
- or throw an argument error telling them to set spawn=true as well.

I think i prefer auto-setting spawn=true based on whether or not it was explicitly set? That seems like the friendliest to me.
So the logic would be something like:
```julia
if threadpool === nothing
    threadpool = :default
else
    spawn = true    
end
```

Reply:
I think i agree: setting `threadpool` should imply `spawn=true`

Another alternative interface would be to just re-use the existing `spawn` keyword, so we support `spawn=true` (spawn in default threadpool), `spawn=:default` (same), `spawn=:interactive` (spawn in the interactive threadpool)

But i'm not sure i like that more. Probably whatever we decide will become the standard name for such a keyword (if/when other functions need to be told which threadpool to use), so have the keyword be `threadpool` seems a decent choice to me.

Reply:
I followed @NHDaly's suggestion.